### ISSUE TITLE
False positive for RS1024 (Compare symbols correctly) when not comparing symbols

### ISSUE THREAD

**Issue by KrisVandermotten** [dotnet/roslyn-analyzers#4413]
It seems that in some cases, when a method has an overload that takes an `IEqualityComparer<T>`, and one of the generic arguments to that method is `ISymbol` or a type that derives from `ISymbol`, the analyzer assumes that the symbols are being compared. That is not necessarily true.

### Steps To Reproduce

Consider this method:

```cs
ILookup<string, IFieldSymbol> GetNames(IFieldSymbol[] fields) => fields.ToLookup(f => f.Name);
```

### Expected behavior

No symbols are being compared, so no diagnostic is reported.

Note that the method has an overload taking a `IEqualityComparer<string>`, it seems this may confuse the analyzer.

### Actual behavior

The following diagnostic is reported:

    warning RS1024: Compare symbols correctly

**Comment by MisinformedDNA** [comment]
Here's another scenario:

```csharp
public List<IPropertySymbol> SourceProperties { get; }
public List<IPropertySymbol> DestinationProperties { get; }

public IEnumerable<string> GetMatchingProperties()
{
    return SourceProperties.Join(DestinationProperties, p => (p.Name, p.Type.Name), p => (p.Name, p.Type.Name), (p1, p2) => p1.Name);
}
```
